operator/Dockerfile

```diff
@@ -28,7 +28,7 @@ ARG VERSION
 RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager -ldflags="-X github.com/pulumi/pulumi-kubernetes-operator/operator/version.Version=${VERSION}" cmd/main.go
 
 # Build the agent binary
-FROM golang:1.22 AS agent-builder
+FROM --platform=${BUILDPLATFORM} golang:1.22 AS agent-builder
```

Comment:
```suggestion
FROM --platform=${BUILDPLATFORM} golang:1.23 AS agent-builder
```

Reply:
I thought about changing it, but the agent's go.mod is indeed 1.22 so I figured we'd defer to https://github.com/pulumi/pulumi-kubernetes-operator/issues/687